frontend/packages/console-shared/src/components/spotlight/Spotlight.tsx

```diff
@@ -7,9 +7,22 @@ type SpotlightProps = {
   interactive?: boolean;
 };
 
+type HasHiddenAncestor = (el: Element) => boolean;
+
 const Spotlight: React.FC<SpotlightProps> = ({ selector, interactive }) => {
+  const hasHiddenAncestor: HasHiddenAncestor = (el) => {
+    if (!el) return false;
+    let element = el;
+    while (element?.parentElement) {
+      const ariaHidden = element?.getAttribute('aria-hidden');
+      if (ariaHidden) return ariaHidden === 'true';
+      element = element.parentElement;
+    }
+    return false;
+  };
+
   const element = React.useMemo(() => document.querySelector(selector), [selector]);
-  if (!element) return null;
+  if (!element || hasHiddenAncestor(element)) return null;
```

Comment:
If your `useMemo` calculated the `element` correctly, no additional check is required.
```suggestion
  if (!element) return null;
```